### DISCUSSION TITLE
How can I set an option as selected when editing a record?

### DISCUSSION THREAD

**Discussion by vulct174** [filamentphp/filament#15185] [Help]
### Package

Form builder

### Package Version

v3.2.116

### How can we help you?

I want the value of the advertiser to be preselected when I edit a record. 
My current code is not working as expected. 

Thank you very much!

![selected](https://github.com/user-attachments/assets/8ce2842c-d1bd-4031-b2a3-4f4dd97c6a35)

**Comment by leandrocfe** [comment]
Use `formatStateUsing`
```php
Select::make('advertiser_ids')
    ->options([
        '1' => 'Advertiser 1',
        '2' => 'Advertiser 2',
        '3' => 'Advertiser 3',
    ])
    ->multiple()
    ->formatStateUsing(function (?array $state, string $operation): ?array {
        if ($operation === 'create' || $operation === 'view') {
            return $state;
        }

        return filled($state) ? $state : ['1'];
    })
```

**Reply by vulct174** [reply]
Hi @leandrocfe, thanks for your reply. It works with the sample code you gave me, but with options taken from query like mine it doesn't seem to work, can you test it?

**Reply by leandrocfe** [reply]
If the sample works, you should apply it according to your model. We cannot see what is being returned in the Select options or the complete output of your code. You should inspect it and apply it as needed.

**Reply by vulct174** [reply]
Thank you for your answer!

**Comment by vulct174** [comment]
Hello @danharrin, Can you please check it for me? Thank you.
